packages/kbn-i18n/src/react/provider.tsx

```diff
@@ -29,12 +29,26 @@ import * as i18n from '../core';
  * IntlProvider should wrap react app's root component (inside each react render method).
  */
 export class I18nProvider extends React.PureComponent {
-  public static propTypes = {
-    children: PropTypes.object,
-  };
+  public static propTypes = { children: PropTypes.element.isRequired };
+  public static contextTypes = { intl: intlShape };
+  public static childContextTypes = { intl: intlShape };
+
+  public getChildContext() {
+    if (this.context.intl && i18n.isPseudoLocale()) {
+      const formatMessage = this.context.intl.formatMessage;
+      this.context.intl.formatMessage = (...args: any[]) => {
+        return i18n.translateUsingPseudoLocale(formatMessage.apply(this.context.intl, args));
+      };
+    }
+
+    return { intl: this.context.intl };
+  }
 
   public render() {
-    const { children } = this.props;
+    const child = React.Children.only(this.props.children);
+    if (this.context.intl) {
```

Comment:
__nit:__ please leave a comment here as well that explains that we can have `IntlProvider` somewhere within ancestors so we just reuse it and don't recreate with another `IntlProvider`.